lib/build.js

```diff
@@ -42,6 +42,11 @@ function build(clusterWorkerId) {
     }
     str = str + encode(alphabet.lookup, seconds);
 
+    // rebuild if str doesn't end in alpha-numeric character
+    while ( !/[a-zA-Z0-9]$/.test(str) ) {
```

Comment:
If you don't like the last character you could always add characters to the end. 

```js
const id = shortid() + 'A';
```

This will be more efficient than re-running the generator.

Reply:
Appending a character to the end breaks the guarantee that every shortid is unique, unless that character is in a position in the shuffled alphabet table that is a multiple of 16.  An easier way to handle this (though it may just a bit compromise the order of the shuffled alphabet) is to ensure that - and _ are in positions in the alphabet table that are a multiple of 16.

Reply:
If it is done systemically and not conditionally, then it doesn't break the guarantee, but the shortid is longer as a result.